### PR TITLE
[New JS Function] Make a number of small improvements

### PR DESCRIPTION
**Pull request by nex3**

### CODE REVIEW

File: proposal/new-function-and-values-api.d.ts [hunk 1]
```diff
@@ -263,7 +278,7 @@ export abstract class Value {
    * Must be the same for `Value`s that are equal to each other according to the
    * `==` SassScript operator.
    */
-  hashCode(): string;
+  hashCode(): number;
```
Comment (Awjin) [inline proposal/new-function-and-values-api.d.ts:281]: We could also change this to a getter.

**Review by Awjin — CHANGES_REQUESTED** [review]
Another improvement that could be worth making here—`convert/coerceValueToMatch` can be more simply defined as returning the value of the result of `convert/coerceToMatch`

File: proposal/new-function-and-values-api.d.ts (hunk 1 above)
Comment (nex3) [reply proposal/new-function-and-values-api.d.ts:281]: I think that violates the `immutable` package protocol.

**Review by nex3 — COMMENTED** [review]
> Another improvement that could be worth making here—`convert/coerceValueToMatch` can be more simply defined as returning the value of the result of `convert/coerceToMatch`

Done.

**Review by Awjin — APPROVED** [review]

**Pull request merged**